### ISSUE TITLE
Debug `ERR_BLOCKED_BY_ADBLOCKER`

### ISSUE THREAD

**Issue by ihucos** [ihucos/simple-web-analytics.com#6]
A user complained about this error on his browser.
```
(index):173 GET https://simple-web-analytics.com/track?referrer=&site=XXXXXX&utcoffset=0 net::ERR_BLOCKED_BY_ADBLOCKER
(anonymous) @ (index):173
(index):1 Uncaught (in promise) TypeError: Failed to fetch
```
I'd need to inspect that and know how many user's could be affected.

**Comment by pyprism** [comment]
Just sharing my experience, simple-web-analytics.com and  counter.dev both domains are blocked by many adblockers. I am using nextdns for adblocking.

**Comment by xplosionmind** [comment]
The same happens with Safari…

@ihucos may I know why did you add a `wontfix` label?

![image](https://user-images.githubusercontent.com/42862428/151929929-418062a4-22b6-474b-ba71-24e4a3757caf.png)

**Comment by ihucos** [comment]
Wontfix because there is nothing really that can be done about it. The client (browser) interprets what the server gives it and it's basically the client's business how to interpret it. So there are ways to make it more difficult for the client to discover it is being tracked by specific javascript code but it's a game I don't want to play because the client has the upper hand and can always adapt to a different signature to detect the tracking code or tracking endpoint.

An approach where that client can not omit tracking is log analysis, there are projects that use this different approach.

To say a little about self hosting. I imagine an adblocker could track instead of `counter.com/track` something along the lines of filtering out all requests to `/track` that contain the used parameters there like also matching for `/screen=/.*?x.*?/` or something similar. 

Another random point: we try to be privacy aware, from that perspective respecting the end-users wish to not be tracked seems like the sensible option.

That is my take on it but open to change my mind ;-)

**Comment by ihucos** [comment]
> The problem I have a lot of reservations in completely switching to counter.dev if [with Plausible I get a total count of 327 visitors](https://plausible.io/tommi.space) last month, that is several times more the [58 logged by counter.dev](https://counter.dev/dashboard.html?user=tommi&token=wTbit2b2U11fYZsE).

That is quite a difference. I'd imagine there must be another factor, so yes would be interesting to cross reference with a third analytics provider to see who is wrong and who is right. If the browser is blocking tracking to counter.dev I would assume it's also blocking tracking via Plausible since it is a bit more popular.

> enforcing ways to bypass blockers are legitimate.

Super interesting argument. From a business perspective that I have to have since I wish to at some point in time being able to work full time on the project the answer is simple: We are providing value to the webmasters, not our webmasters users. So yes, of course we should try to circumvent adblockers. Using your own domain is really totally fine, I just don't want to implement anything that would be too invasive. Then we "just" need to prioritize and implement with the given time resource :sweat_smile:

**Comment by xplosionmind** [comment]
> I would assume it's also blocking tracking via Plausible since it is a bit more popular.

It does not, since Plausible offers the possibility of using one’s own domain…

**Comment by ihucos** [comment]
> It does not, since Plausible offers the possibility of using one’s own domain…

Gotcha, I would not think that the results would differ so much because of that reason. Yeah, it's definitely relevant.

**Comment by Tiger862000** [comment]
Hi, first of all thank you for even providing PWYW privacy friendly analytics solution (product as a service). You are a hero to me @ihucos .  

I just started to use the service and consider finacially contributing if it will serve my needs...
Coming to the actual topic:
I wanted to know why this topic was closed although there is no fix for it? 

Before you answer hear me out. The problem is that many browsers nowadays use adblockers and thus simply block counter.dev, and the figures are increasing. Having that said, sooner or later it will render the service useless if at some point in future then even my MUM's browser will use an integrated adblocker. You get the point...

So what are the options here? 
I understood that using custom domains may be a workaround. A solution actually also the big players utilize. Why not going down that path? 

Couldn't that be a PRO feature with that you can start making money? I have read your blog post about PWYW on your website and I don't like it. I will explain a bit later. 
Beforehand let's talk business model: While PRO still would mean PWYW, but the difference is that a user would have to pay at least 1€/mo (and more if they like). Which will increase your income (today 27ct/mo/user) to significant higher figure. 
If you go down that road I will be your first customer (could imagine paying 2-3 €/mo). 
Why I don't like your blog post about the status of your business: My due dilligence on your service included next to privacy aspect also of course your business model and yes I like the PWYW concept, but reading that you earn s**t about it and that the project is underfinaced, imediately it gave me the idea to drop using your solution once I can afford an effective and fucntioning solution. @ihucos as initially said "you are a hero" to me providing this hosted solution, so let me give you an advice from business owner to business owner, having a stable income stream is substancial for every product / business, without ressources the business will die and with it the product (as a service as you provide it). 
I invite you to PM me to talk more about a fair business model. 

Honest greets from Austria!
KR, Stefan

**Comment by ihucos** [comment]
Hello Stefan,

thanks you for the very long message and interesting point. I'd be very happy to discuss some aspects privately, can you provide me your mail? Or write me at mail@irae.me :-)

Regarding a custom domain: The ticket was closed and then re-opened since I changed my mind. This feature will be implemented, it's just a little tricky as I am dependant on Cloudflare at the moment and may need to not use this service. But I get your point and understand the feedback from the other users as wee. In the end of they day all I am doing is serving my users and while still having an own opinion I don't want to be too big-headed about anything. 

Regarding the PWYW. While I get a continuous flow of new integrations, having the only Free and at the same arguably privacy friendly web analytics product did not get really substantial traction yet. I will not ditch this business model out yet and I would not feel to good changing the unwritten PWYW terms that current users signed up with. But something I am currently working on is to better communicate to active users that that a service fee is expected. See this PR with it's screenshot: https://github.com/ihucos/counter.dev/compare/paytest?expand=1

> [...] once I can afford an effective and fucntioning solution.

Do you imply a non-functioning product following from the business model or do you mean this open issue or feature request by that. The project is so far two years old, so are the oldest active users and if anything is not working for you on a technical level let me know and it will be fixed. That being said, yes, it's not easy running something like this next to other things but from a purely technical and functional perspective I think it's just a great product. This open issue will get addressed before it burns :-)


> Having that said, sooner or later it will render the service useless if at some point in future then even my MUM's browser will use an integrated adblocker. You get the point...

I agree. Pitty for many users changing DNS setting are not really doable but docs could help.

**Comment by ihucos** [comment]
The good news is that using a proxy server is available right now as it does not require much work from the web analytics providers side.

What the script already supports is a `data-server` attribute, example:

```
<script src="https://cdn.counter.dev/script.js" data-id="myuuid" data-utcoffset="1" server="https://example.com"></script>`
```

So you'd need in the end to proxy those endpoints

```
proxy/track -> counter.dev/track ->
proxy/trackpage -> counter.dev/trackpage
proxy/script.js -> cdn.counter.dev/script.js
```
In that case the tracking script inclusion would look like this:

```
```
<script src="https://proxy/script.js" data-id="myuuid" data-utcoffset="1" server="https://proxy"></script>`
```

**Comment by adian** [comment]
@ihucos There is a typo in your last comment. There should be `data-server`, not `server`.

**Comment by ihucos** [comment]
@adian thanks. typo fixed.

**Comment by ihucos** [comment]
There was a nice pull request related to a proxy setup: https://github.com/ihucos/counter.dev/pull/102
